### ISSUE TITLE
Mounting images via UDisks2

### ISSUE THREAD

**Issue by gasinvein** [brezerk/q4wine#184]
UDisks is a privileged daemon with D-Bus API for for manipulating block devices and filesystems.
Q4Wine could ask UDisks2 to mount a disk image for it. This gives several advantages over exec'ing CLI tools for mount:
- Q4Wine won't need to elevate at all, even no need to run suid binaries (e.g. `fusermount` when using FUSE)
- UDisks2 is preinstalled on most modern Linux desktops, so no additional software required
- This can work in sandbox with NO_NEW_PRIVILEGES set (useful for Flatpak, #147)

Key difference from current mount approach, as I see it, is that UDisks2 sets up a mountpoint for us automatically (usually in `/media` or `/run/media/`) when mounting a filesystem, which means we can't select it ourselves.
To mount an image, client app would need to call two D-Bus methods sequentially: `org.freedesktop.UDisks2.Manager.LoopSetup` to create a block device backed by given image (returns object path of the created device), and then `org.freedesktop.UDisks2.Filesystem.Mount` (method of the block device object we just created) to mount the filesystem (returns path to the mount point).

**Comment by brezerk** [comment]
as per: https://wiki.archlinux.org/title/udisks
seems like there are two ways to interact with udisks: dbus and cli.
cli probably would be much easier to implement.

**Comment by brezerk** [comment]
or seems like there is much interesting way: https://doc.qt.io/qt-5/qdbusinterface.html

**Comment by brezerk** [comment]
api reference: http://storaged.org/doc/udisks2-api/latest/

**Comment by gasinvein** [comment]
Yeah, we don't need much of the API here, just two interfaces and two method calls. Probably the only thing to note is that when calling `LoopSetup` method on `org.freedesktop.UDisks2.Manager` interface, we should set `no-part-scan` option to `true`, so the filesystem interface appears for the whole loopback block device and we won't need to poke into partitions to mount the filesystem.
